internal/bgp/frr/frr.go

```diff
@@ -286,8 +286,10 @@ func (sm *sessionManager) createConfig() (*frrConfig, error) {
 			switch family {
 			case ipfamily.IPv4:
 				rout.ipV4Prefixes[prefix] = prefix
+				neighbor.HasV4Advertisements = true
```

Comment:
It's the same as appending to advertisements. This flag is a shortcut to avoid looping over advertisements in order to understand if we need to add the deny any section or not, but it is not adding any information per se.
The map is being built every time we call this function, and what we care about is whether the peer has at least one v4 (and v6) advertisement in order to do the proper rendering